packages/p2p-v2/src/pages/my-ads/screens/CreateEditAd/__tests__/CreateEditAd.spec.tsx

```diff
@@ -165,11 +165,22 @@ jest.mock('@deriv/api-v2', () => ({
     useActiveAccount: () => ({ data: { currency: 'USD' } }),
 }));
 
-jest.mock('@/hooks', () => ({
-    ...jest.requireActual('@/hooks'),
-    useFloatingRate: () => ({ rateType: 'floating' }),
-    useQueryString: jest.fn().mockReturnValue({ queryString: { advertId: '' } }),
-}));
+jest.mock('@/hooks', () => {
+    const modalManager = {
+        hideModal: jest.fn(),
+        isModalOpenFor: jest.fn(),
+        showModal: jest.fn(),
+    };
+    modalManager.showModal.mockImplementation(() => {
+        modalManager.isModalOpenFor.mockReturnValue(true);
+    });
+    return {
+        ...jest.requireActual('@/hooks'),
+        useFloatingRate: () => ({ rateType: 'floating' }),
+        useModalManager: jest.fn().mockReturnValue(modalManager),
+        useQueryString: jest.fn().mockReturnValue({ queryString: { advertId: '' } }),
+    };
```

Comment:
a much more straight forward way to mock modal manager is like this:
```suggestion
        ...jest.requireActual('@/hooks'),
        useFloatingRate: () => ({ rateType: 'floating' }),
        useModalManager: jest.fn().mockReturnValue({
            hideModal: jest.fn(),
            isModalOpenFor: jest.fn(),
            showModal: jest.fn(),
        }),
        useQueryString: jest.fn().mockReturnValue({ queryString: { advertId: '' } }),
```

Reply:
this would work, but then calling the showmodal wont open the modal .

Reply:
hmmm thats weird cause it was ok for me, but if it passes then thats fine

Reply:
can resolve my other comments 👍